pkg/reboot/rebooter_test.go

```diff
@@ -0,0 +1,64 @@
+package reboot
+
+import (
+	"context"
+	"os"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/medik8s/self-node-remediation/pkg/utils"
+	"github.com/medik8s/self-node-remediation/pkg/watchdog"
+)
+
+var isSoftwareRebootCalled bool
+
+var _ = Describe("Watchdog tests", func() {
+	var rebooter *watchdogRebooter
+
+	Context("Crash on start", func() {
+		BeforeEach(func() {
+			wd, _ := watchdog.NewFake(false)
+			rebooter = &watchdogRebooter{wd, ctrl.Log.WithName("fake rebooter"), fakeSoftwareReboot}
+
+		})
+
+		AfterEach(func() {
+			isSoftwareRebootCalled = false
+		})
+
+		Context("Software reboot is disabled", func() {
+			It("watchdog should not start", func() {
+				wd := rebooter.wd
+				err := wd.Start(context.TODO())
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("failed to start watchdog, can't default to software reboot"))
+				Expect(wd.Status()).To(Equal(watchdog.Disarmed))
+			})
+		})
+
+		Context("Software reboot is enabled", func() {
+			BeforeEach(func() {
+				_ = os.Setenv(utils.IsSoftwareRebootEnabledEnvVar, "true")
+			})
+			It("should return healthy", func() {
+				wd := rebooter.wd
+				err := wd.Start(context.TODO())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(wd.Status()).To(Equal(watchdog.Malfunction))
+				//Verify reboot goes as expected
+				Expect(rebooter.Reboot()).ToNot(HaveOccurred())
+				Expect(isSoftwareRebootCalled).To(BeTrue())
+			})
+		})
+
+	})
```

Comment:
NIT: I am not so much of an expert in UT but is it ok to use `Context` inside `Context`? By looking at other examples, like in https://onsi.github.io/ginkgo/, I don't see Context block inside another Context block. I think `Describe` might be better suited.
`Context("Crash ...` -> `Describe("Crash ...`